server/src/main/java/org/opensearch/rest/action/search/RestSearchAction.java

```diff
@@ -223,6 +224,9 @@ public static void parseSearchRequest(
         }
 
         searchRequest.setCancelAfterTimeInterval(request.paramAsTime("cancel_after_time_interval", null));
+
+        // set remote address for searchRequest
+        searchRequest.remoteAddress(new TransportAddress(request.getHttpChannel().getRemoteAddress()));
```

Comment:
I don't think it's safe to hijack `remoteAddress()` like this. 

It's intended to be set by `InboundHandler` on transport requests. Setting it on a REST request may have unintended consequences.

Also, if the request comes through a load balancer or a proxy, this value will almost certainly be the load balancer or proxy's address.